builtin/read.c

```diff
@@ -55,8 +51,36 @@ int builtin_read(struct mrsh_state *state, int argc, char *argv[]) {
 		mrsh_buffer_append_char(&buf, (char)c);
 	}
 	mrsh_buffer_append_char(&buf, '\0');
+	
+	struct mrsh_word_string *str = mrsh_word_string_create(mrsh_buffer_steal(&buf), false);
```

Comment:
Nitpick: is it possible to rename this to `ws`? It's too easy to assume it's just a `char *`.